### ISSUE TITLE
Add IPv6 support for `google_compute_instance`

### ISSUE THREAD

**Issue by msampathkumar** [hashicorp/terraform-provider-google#13916]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Below is from https://paste.googleplex.com/5939776506036224

Currently Terraform support assigning and accessing IPv4 address in `google_compute_instance`. Ask for this request to provide the sample for `IPv6` Addresses.

```tf
resource "google_compute_instance" "ipv6destination" {
  name         = "destination-vm"
  machine_type = "e2-medium"
  zone         = "us-west2-a"
 
  network_interface {
    network      = google_compute_network.ipv6default.id
    subnetwork   = google_compute_subnetwork.ipv6default.id
    stack_type   = "IPV4_IPV6"
    network_ip   = "10.0.0.143"
    network_ipv6 = "001:0db8:85a3:::8a2e:0370:7334"  # this is the required feature
  }
}
```
1. Like shown above, provide provider config option `network_ipv6` for setting an example IPv6 Address
1. provide `network_interface.0.ipv6_access_config.0.interal_ipv6` option to access a VM' internal IPv6 address.

As of now, after enabling internal IPv6 address by using `stack_type = "IPV4_IPV6"`, there is no way to access VM's internal IPv6 Address.

### New or Affected Resource(s)

* google_compute_instance

### References

* https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance

**Comment by rileykarson** [comment]
This looks supported already? https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#stack_type

**Comment by msampathkumar** [comment]
Hi, can you tell me how to access an internal IPv6 address please or can you provide a code sample where IPv6 Address is manually assigned?

**Comment by rileykarson** [comment]
That should be in https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#ipv6_access_config, specifically as https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#network_interface.0.ipv6_access_config.0.external_ipv6?

**Comment by rileykarson** [comment]
Note that there is no field called `internal_ip`, see https://cloud.google.com/compute/docs/reference/rest/v1/instances

I'd consider parity with `ipv6AccessConfigs` the closure criteria here.

**Comment by msampathkumar** [comment]
Hi, Thanks for quick reply. I updated the summary now. If its possible get the internal IPv6 address of VM, then I am content with it.

**Comment by rileykarson** [comment]
Updated the parent to refer to the field by its name in the API (ipv6Address / `ipv6_address`)

**Comment by jlkiri** [comment]
Obtaining instance's internal IPv6  ULA is extremely painful right now. It should be possible to use that information in dependent resources.

**Comment by booya** [comment]
I agree on needing full configuration for the ipv6AccessConfigs. In our case, we need to use reserved IPv6 addresses on compute instances, but the provider does not expose `networkInterfaces[].ipv6AccessConfigs[].externalIpv6`

**Comment by alexandre-rozand** [comment]
I'm feeling the pain, trying to assign a reserved static ipv6 to an instance and currently not possible via TF.

**Comment by jeremiahlee** [comment]
I first thought this was a bug, but realized it is a missing feature when examining the debug information.

I created a project to demonstrate this issue. https://github.com/jeremiahlee/bug-tf-gcp-reserved-ipv6/

If you are trying to use a reserved, regional, IPv6 address for a `google_compute_instance`, it will produce an error like:

`> Error: Error creating instance: googleapi: Error 400: Invalid value for field 'resource.networkInterfaces[0].accessConfigs[1].natIP': '2600:1900:4150:ba95::'. The specified external IP address '2600:1900:4150:ba95::' was not found in region 'europe-north1'., invalid`

When [comparing what Terraform tries with what the GCP console would do](https://github.com/jeremiahlee/bug-tf-gcp-reserved-ipv6/blob/main/debug-hypothesis.md), we see that `ipv6_access_config` needs to be expanded to include `externalIpv6` and `externalIpv6PrefixLength`. (For now, `"type": "DIRECT_IPV6"` [is the only supported option](https://cloud.google.com/compute/docs/reference/rest/v1/instances) and do not need to be included, just set.)

The potential configuration could look like:

```tf
resource "google_compute_instance" "default" {
  network_interface {
    stack_type = "IPV4_IPV6"
    network    = google_compute_network.test_vpc.self_link
    subnetwork = google_compute_subnetwork.test_external_subnet.self_link

    access_config {
      nat_ip       = "34.88.182.23"
      network_tier = "PREMIUM"
    }

    ipv6_access_config {
      external_ipv6               = "2600:1900:4150:ba95::" # New!
      external_ipv6_prefix_length = 96 # New!
      network_tier                = "PREMIUM"
    }
  }
}
```

**Comment by msampathkumar** [comment]
Hi @jeremiahlee  Sound great! Let me try this.
